src/game/savegame/savegame.c

```diff
@@ -544,29 +552,33 @@ void Savegame_ScanSavedGames(void)
     }
 
     REQUEST_INFO *req = &g_SavegameRequester;
+    Requester_ClearTextstrings(req);
+    char savegame_text[MAX_LEVEL_NAME_LENGTH] = { 0 };
+    uint16_t flags = 0;
 
-    req->items = 0;
-    for (int i = 0; i < g_Config.maximum_save_slots; i++) {
+    for (int i = 0; i < req->max_items; i++) {
         SAVEGAME_INFO *savegame_info = &m_SavegameInfo[i];
 
         if (savegame_info->level_title) {
-            req->item_flags[req->items] &= ~RIF_BLOCKED;
-
-            sprintf(
-                &req->item_texts[req->items * req->item_text_len], "%s %d",
-                savegame_info->level_title, savegame_info->counter);
-
             if (savegame_info->counter == g_SaveCounter) {
                 m_NewestSlot = i;
             }
+
+            flags = req->item_flags[req->items_used] & ~RIF_BLOCKED;
+            size_t out_size = snprintf(
+                                  NULL, 0, "%s %d", savegame_info->level_title,
+                                  savegame_info->counter)
+                + 1;
+            snprintf(
+                savegame_text, out_size, "%s %d", savegame_info->level_title,
+                savegame_info->counter);
```

Comment:
Here we use an intermediate buffer `savegame_text`, but the same principles apply, only with `MAX_LEVEL_NAME_LENGTH`.

Reply:
Yes I did it this way so we could use `Requester_AddItem` instead of editing the requestor itself directly. The if/else use different string formats. But if we make the change to your format idea in `Requester_AddItem`, that would eliminate the need to do this. So I'll make that change.